pytensor/tensor/math.py

```diff
@@ -2366,7 +2361,92 @@ def tensordot(a, b, axes=2):
     See the documentation of numpy.tensordot for more examples.
 
     """
-    return _tensordot_as_dot(a, b, axes, dot=dot, batched=False)
+    try:
+        iter(axes)
+    except Exception:
+        axes_a = list(range(-axes, 0))
+        axes_b = list(range(0, axes))
+    else:
+        axes_a, axes_b = axes
+    try:
+        na = len(axes_a)
+        axes_a = list(axes_a)
+    except TypeError:
+        axes_a = [axes_a]
+        na = 1
+    try:
+        nb = len(axes_b)
+        axes_b = list(axes_b)
+    except TypeError:
+        axes_b = [axes_b]
+        nb = 1
+
+    a = as_tensor_variable(a)
+    b = as_tensor_variable(b)
+    as_ = a.shape
+    bra = a.broadcastable
+    ats = a.type.shape
+    nda = a.ndim
+    bs = b.shape
+    brb = b.broadcastable
+    bts = b.type.shape
+    ndb = b.ndim
+    if na != nb:
+        raise ValueError(
+            "The number of axes supplied for tensordot must be equal for each tensor. "
+            f"Got {na} and {nb} respectively."
+        )
+    for k in range(na):
+        ax_a = axes_a[k]
+        ax_b = axes_b[k]
+        if ax_a < 0:
+            axes_a[k] += nda
+        if axes_a[k] < 0 or axes_a[k] >= nda:
```

Comment:
Une numpy utility to normalize axis? It takes care of out of bounds and raises for repeated axes.

`normalize_axis_index` or `normalize_axis_tuple` (something like that I never remember the name, it's used in several places in the codebase)